packages/block-library/src/buttons/style.scss

```diff
@@ -116,7 +75,7 @@ $blocks-block__margin: 0.5em;
 	/* stylelint-enable indentation */
 		margin-left: auto;
 		margin-right: auto;
-		margin-bottom: $blocks-block__margin;
+		margin-bottom: calc(var(--wp--style--block-gap, #{ $blocks-block__margin}) * 0.5);
```

Comment:
Same note as above, it looks like there isn't a corresponding `margin-top` so this margin appears to be half-height?